Automattic/jetpack
# Contact Form: Enable undoing adding a contact form

See https://github.com/Automattic/jetpack/issues/35505

## Proposed changes:

Adding a Contact Form block in the Gutenberg editor cannot be undone. This PR fixes it.

This issue is due to some attributes being modified with the `setAttributes` function not after a user interaction but rather on mount inside a `useEffect` hook though I don't understand why that's a problem. 3 cases needed to be fixed:

1. The `to` and `subject` attributes. Instead of setting their default value client-side in a `useEffect` hook, we now get the values server-side, pass them to the client via the `window` object, and retrieve them when registering the block client-side.
2. The `requiredText` attribute. Actually, it's not in the list of block attributes, so instead of considering it as is, we simply set a fallback value when rendering a field label.
3. The `formTitle` attribute. This attribute is used to provide an accessible name label to a form for screen readers. Its computation is a bit computation-heavy, and I've decided to remove it completely for now instead of implementing a server-side version that could slow down the rendering of the page.

### Other information:


## Jetpack product discussion


n/a

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






- Spin up a test site with this branch
- Build the `forms` package and the Jetpack plugin if you're running this locally
- Create a new post
- Add a _Form_ block
- Undo the action and check it's been removed
- Add a _Contact Form_ block
- Undo the action and check it's been removed
- Add it back
- In the _Email Connection_ panel of the sidebar, verify that the email recipient and subject have a default value
<img width="288" alt="Screenshot 2024-03-20 at 3 43 18 PM" src="https://github.com/Automattic/jetpack/assets/1620183/af221cdc-42cc-4e85-9a6d-bbd16793cee9">
